daemon/src/projection.rs

```diff
@@ -694,16 +710,46 @@ pub fn to_mempool_url(txid: Txid, network: Network) -> String {
     }
 }
 
+/// Link to transaction on mempool.space for UI representation
+#[derive(Debug, Clone, Serialize)]
+struct TxUrl {
+    pub label: TxLabel,
+    pub url: String,
+}
+
 impl TxUrl {
-    pub fn new(txid: Txid, network: Network, label: TxLabel) -> Self {
+    fn new(txid: Txid, network: Network, label: TxLabel) -> Self {
         Self {
             label,
             url: to_mempool_url(txid, network),
         }
     }
+
+    /// Highlight particular transaction output in the TxUrl
+    fn with_output_index(mut self, index: u32) -> Self {
+        self.url.push_str(&("#".to_string() + &index.to_string()));
+        self
+    }
+
+    /// If the Transaction contains the script_pubkey, output will be selected
+    /// in the URL. Otherwise, fall back to the main txid URL.
+    fn from_transaction(
+        transaction: &Transaction,
+        script_pubkey: &Script,
+        network: Network,
+        label: TxLabel,
+    ) -> Self {
+        debug_assert!(label != TxLabel::Commit, "commit transaction has a single output which does not belong to either party - this won't highlight anything");
```

Comment:
:+1: Haha, nice! I never use `debug_assert`s.

Reply:
To be fair, we are highlighting an output that doesn't belong to any one party in the lock transaction too, so you could just say: `commit transaction only ever has one output, nothing to highlight`.

Reply:
I got confused - why are we highlighting the one in lock transaction then, and not this one too?

Reply:
I kept the old wording, otherwise the debug_assert makes less sense. the lock one has a comment next to the invocation.

Reply:
> I got confused - why are we highlighting the one in lock transaction then, and not this one too?

If there's only one output do you need to highlight anything?

At least in the context of the lock transaction you can highlight the one output that matters for the protocol, even if you can't also show the change outputs that belong to that specific party.